src/pubd/repository.rs

```diff
@@ -931,13 +931,9 @@ impl RepositoryAccessProxy {
     }
 
     fn read(&self) -> KrillResult<Arc<RepositoryAccess>> {
-        match self.store.get_latest(&self.key) {
-            Ok(repo) => Ok(repo),
-            Err(e) => match e {
-                AggregateStoreError::UnknownAggregate(_) => Err(Error::RepositoryServerNotEnabled),
-                _ => Err(Error::AggregateStoreError(e)),
-            },
-        }
+        self.store
+            .get_latest(&self.key)
+            .map_err(|_| Error::custom("Publication Server data missing"))
```

Comment:
What would the actual error here contain and should it be logged? Should self.key be logged? Is it correct that this uses a custom error instead of typed error?

Reply:
Yeah, you're right.. not sure why I left it out. I pushed a fix to include the original error message. This case really should never happen, unless there is some data corruption issue.